src/assets/scripts/client/InputController.js

```diff
@@ -332,104 +268,82 @@ export default class InputController {
-            event.preventDefault();
-        } else if (event.which === KEY_CODES.PAGE_DOWN) {
-            this._tutorialView.tutorial_next();
-            event.preventDefault();
-        }
+        // // For firefox see: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
+        // const is_firefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+        //
+        // if (!GameController.game_paused()) {
+        //     this.$commandInput.focus();
+        // }
+        //
+        // if (event.which === KEY_CODES.ESCAPE) {
+        //     if (prop.tutorial.open) {
+        //         this._tutorialView.tutorial_close();
+        //     } else if ($(SELECTORS.DOM_SELECTORS.AIRPORT_SWITCH).hasClass(SELECTORS.CLASSNAMES.OPEN)) {
+        //         UiController.ui_airport_close();
+        //     }
+        //
+        //     this.selectAircraft();
+        // }
     }
 
+    // TODO: Is this really needed??
     /**
      * @for InputController
-     * @method input_parse
+     * @method onCommandInputChangeHandler
      */
-    input_parse() {
-        this.input.callsign = '';
-        this.input.data = '';
-
-        if (this.input.command.length === 0) {
-            return;
-        }
-
-        // TODO: move to master REGEX constant
-        const match = /^\s*(\w+)/.exec(this.input.command);
-
-        if (!match) {
-            return;
-        }
-
-        this.input.callsign = match[1];
-        prop.canvas.dirty = true;
-
-        // TODO: this looks like it should happen in the `AircraftController`
-        for (let i = 0; i < this._aircraftController.aircraft.list.length; i++) {
-            const aircraft = this._aircraftController.aircraft.list[i];
-
-            if (aircraft.matchCallsign(this.input.callsign)) {
-                this._aircraftController.onSelectAircraftStrip(aircraft);
-
-                break;
-            }
-        }
+    onCommandInputChangeHandler() {
+        this.input.command = this.$commandInput.val();
     }
 
     /**
      * @for InputController
-     * @method onCommandInputChangeHandler
+     * @method selectAircraft
+     * @param aircraftModel {AircraftModel}
      */
-    onCommandInputChangeHandler() {
-        this.tab_completion_reset();
+    selectAircraft = (aircraftModel) => {
+        if (!aircraftModel) {
+            // TODO: Refactor out the prop
+            // using `prop` here so CanvasController knows which aircraft is selected
+            prop.input.callsign = '';
+            prop.input.command = '';
+            this.input.callsign = '';
+            this.input.command = '';
+            this.$commandInput.val('');
+            this._eventBus.trigger(EVENT.DESELECT_ACTIVE_STRIP_VIEW, {});
 
-        this.input.command = this.$commandInput.val();
+            return;
+        }
 
-        this.input_parse();
-    }
+        // TODO: Refactor out the prop
+        // using `prop` here so CanvasController knows which aircraft is selected
```

Comment:
I feel like the `EventBus` would be well suited for that, but that smells like a stinky change to make 🤕